first-app/src/app/create-shopping-list/create-shopping-list.component.html

```diff
@@ -0,0 +1,12 @@
+<h1>My Shopping List</h1>
+<form (ngSubmit)="addItem()" #shoppingForm="ngForm">
+    <label>Enter a new item:</label>
+    <input type="text" [(ngModel)]="newItem" name="itemInput" placeholder="Enter a new item" required>
+    <button type="submit">Add Item</button>
+</form>
+<ul>
+    <li *ngFor="let item of shoppingList">
```

Comment:
@for and take index like let i = $index. then pass the i to deleteItem method. splice directly instead of filter